app/lib/submission_builder/ty2024/states/nj/nj_return_xml.rb

```diff
@@ -102,7 +108,20 @@ def additional_dependents
           end
 
           def calculator 
-            @submission.data_source.tax_calculator
+            calculator = @submission.data_source.tax_calculator
+            calculator.calculate
+            calculator
+          end
+
+          def financial_transaction
+            calculator = @submission.data_source.tax_calculator
+            calculator.calculate
```

Comment:
[pebble] I know this was copied from elsewhere, but rather than call `.calculate` here, it would be better to follow the pattern from AZ and use [calculated_fields](https://github.com/codeforamerica/vita-min/blob/e9eca3906046dd89e1d65e7585daa2e99081d9ca/app/lib/submission_builder/ty2022/states/az/az_return_xml.rb#L223-L225) to reduce the calls to `.calculate`. 

This should be updated in ID, MD, and NC - we could update NJ at the same time we do the others if you'd rather merge and move on.